### ISSUE TITLE
[Bug] Shell routing issue in xamarin forms

### ISSUE THREAD

**Issue by Srikanth540-L** [xamarin/Xamarin.Forms#12413]
Xamarin forms : 4.8 Latest

We are using FlyoutItem where we have 5 tabs in it and 4 menu items
The First tab is Home , if we select any menu item from flyout that page is loaded with the bottom tabs as it's our design requirement.
The Ui shows like bottom tab home is in selected colour and if try to hit the home tab again it**'s not navigating back to Home page in Android but in IOS it navigates** , The expected behaviour should navigate back to home in Android too.

**Comment by jsuarezruiz** [comment]
@Srikanth540-L Could you attach a small example where reproduce the issue?

**Comment by Srikanth540-L** [comment]
[Xamarin_Forms___Xaminals.zip](https://github.com/xamarin/Xamarin.Forms/files/5389813/Xamarin_Forms___Xaminals.zip)
@jsuarezruiz   I have updated the scenario in Xamarin_Forms___Xaminals sample, please consider the tab which shows monkeys and select any monkey information displayed in the list which routes to detail page and click any other tab and then route back to monkey tab , there we can notice that we are detail page of monkey rather than the monkeys tab page

**Comment by PureWeen** [comment]
@Srikanth540-L can you attach a short video? I'm not able to reproduce the issue you are describing

**Comment by Srikanth540-L** [comment]
@PureWeen The scenario is simple as described above we select monkeys tab and go to any monkeys details and from there we select any other tab and select monkey tab we will land on monkey detail which has back button on top left and tabs below, In our main application we don't have back button since we have back button in the sample you can navigate back to monkeys tab, The question is in ios it take 2 clicks on monkey tab to go the monkeys list screen where as in android it's not ,So we would like to know if it's the default behaviour in Android, Is it must to have the back button to navigate back or we need a suggestion where to handle to navigate back to monkeys tab irrespective of detail page is between the monkey's list page. if we click on monkey tab I hope you got my view.

**Comment by jsuarezruiz** [comment]
@Srikanth540-L I can reproduce the behavior you commented. I think it would be important to have the same behavior on all platforms and in this case, the behavior is different on Android and iOS.

**Comment by Srikanth540-L** [comment]
@jsuarezruiz  Exactly, The behavior on the all the platforms should be same if the same tab is also the part of flyout it generates the proper route , I believe the same should happen for tabs
